sdk/digitaltwins/digital-twins-parser/review/digital-twins-parser.api.md

```diff
@@ -0,0 +1,507 @@
+## API Report File for "@azure/digital-twins-parser"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+// @public (undocumented)
+export interface ArrayInfo extends ComplexSchemaInfo {
+    // (undocumented)
+    elementSchema?: SchemaInfo;
+    // (undocumented)
+    entityKind: "array";
+}
+
+// @public (undocumented)
+export interface BooleanInfo extends PrimitiveSchemaInfo {
+    // (undocumented)
+    entityKind: "boolean";
+}
+
+// @public (undocumented)
+export interface CommandInfo extends ContentInfo {
+    // (undocumented)
+    commandType?: CommandTypeInfo;
+    // (undocumented)
+    entityKind: "command";
+    // (undocumented)
+    request?: CommandPayloadInfo;
+    // (undocumented)
+    response?: CommandPayloadInfo;
+}
+
+// @public (undocumented)
+export interface CommandPayloadInfo extends SchemaFieldInfo {
+    // (undocumented)
+    entityKind: "commandpayload" | "commandrequest" | "commandresponse";
+}
+
+// @public (undocumented)
+export interface CommandRequestInfo extends CommandPayloadInfo {
+    // (undocumented)
+    entityKind: "commandrequest";
+}
+
+// @public (undocumented)
+export interface CommandResponseInfo extends CommandPayloadInfo {
+    // (undocumented)
+    entityKind: "commandresponse";
+}
+
+// @public (undocumented)
+export interface CommandTypeInfo extends EntityInfo {
+    // (undocumented)
+    entityKind: "commandtype";
+}
+
+// @public (undocumented)
+export interface ComplexSchemaInfo extends SchemaInfo {
+    // (undocumented)
+    entityKind: "array" | "enum" | "map" | "object";
+}
+
+// @public (undocumented)
+export interface ComponentInfo extends ContentInfo {
+    // (undocumented)
+    entityKind: "component";
+    // (undocumented)
+    schema?: InterfaceInfo;
+}
+
+// @public (undocumented)
+export interface ContentInfo extends NamedEntityInfo {
+    // (undocumented)
+    entityKind: "command" | "component" | "property" | "relationship" | "telemetry";
+}
+
+// @public
+export function createParser(parsingOptions: ModelParsingOption): ModelParser;
+
+// @public (undocumented)
+export interface DateInfo extends TemporalSchemaInfo {
+    // (undocumented)
+    entityKind: "date";
+}
+
+// @public (undocumented)
+export interface DateTimeInfo extends TemporalSchemaInfo {
+    // (undocumented)
+    entityKind: "datetime";
+}
+
+// @public (undocumented)
+export interface DoubleInfo extends NumericSchemaInfo {
+    // (undocumented)
+    entityKind: "double";
+}
+
+// @public (undocumented)
+export type DtmiResolver = (identifiers: string[]) => Promise<string[] | null>;
+
+// @public (undocumented)
+export interface DurationInfo extends TemporalSchemaInfo {
+    // (undocumented)
+    entityKind: "duration";
+}
+
+// @public (undocumented)
+export interface EntityInfo {
+    // (undocumented)
+    childOf: string | undefined;
```

Comment:
I'm curious why this is ` childOf: string | undefined` instead of `childOf?: string`?